phoenix-core/src/main/java/org/apache/phoenix/mapreduce/MultiHfileOutputFormat.java

```diff
@@ -122,11 +123,11 @@ public RecordWriter<TableRowkeyPair, Cell> getRecordWriter(TaskAttemptContext co
      * @return
      * @throws IOException 
      */
-    static <V extends Cell> RecordWriter<TableRowkeyPair, V> createRecordWriter(final TaskAttemptContext context)
+    static <V extends Cell> RecordWriter<TableRowkeyPair, V> createRecordWriter(
+        final TaskAttemptContext context, final OutputCommitter committer)
             throws IOException {
         // Get the path of the temporary output file
-        final Path outputPath = FileOutputFormat.getOutputPath(context);
-        final Path outputdir = new FileOutputCommitter(outputPath, context).getWorkPath();
+        final Path outputdir = ((PathOutputCommitter) committer).getOutputPath();
```

Comment:
this is no longer a h path, it is the final path of the work, the same value as  L128 in the original code.

Reply:
@ss77892 

This indeed looks incorrect.
We should use .getWorkPath() here.

This works for the Magic S3a committer where the work and output path are the same, but for FileCommitter (i.e. HDFS) and others this breaks the commit mechanism by writing into the output directory directory directly.